src/Runner.Common/JobServer.cs

```diff
@@ -125,6 +157,46 @@ private void CheckConnection()
             }
         }
 
+        private void InitializeWebsocketClient(TimeSpan delay)
+        {
+             if (_serviceEndpoint.Authorization != null &&
+                 _serviceEndpoint.Authorization.Parameters.TryGetValue(EndpointAuthorizationParameters.AccessToken, out var accessToken) &&
+                 !string.IsNullOrEmpty(accessToken))
+            {
+                if (_serviceEndpoint.Data.TryGetValue("FeedStreamUrl", out var feedStreamUrl) && !string.IsNullOrEmpty(feedStreamUrl))
+                {
+                    // let's ensure we use the right scheme
+                    feedStreamUrl = feedStreamUrl.Replace("https://", "wss://").Replace("http://", "ws://");
+                    Trace.Info($"Creating websocket client ..." + feedStreamUrl);
+                    this._websocketClient = new ClientWebSocket();
+                    this._websocketClient.Options.SetRequestHeader("Authorization", $"Bearer {accessToken}");
```

Comment:
What's the different between `SetRequestHeader` vs set `Options.Credentials`?
I saw there is a Proxy property under the `ClientWebSocket.Options`, do we need to set that?

Reply:
`Credentials` is essentially cookie based ones etc., it doesn't cover bearer.

I don't think we need to set `proxy` property, I tested this using proxy and it worked fine.